databricks_cli/stack/api.py

```diff
@@ -0,0 +1,329 @@
+        click.echo("Job Link: %s" % job_link)
+        physical_id = {'job_id': job_id, "link": job_link}
+        deploy_output = self.jobs_client.get_job(job_id)
+        return physical_id, deploy_output
+
+    def deploy_resource(self, resource):  # overwrite to be added
+        """
+        Deploys a resource given a resource information extracted from the stack JSON configuration
+        template.
+
+        :param resource: A dict of the resource with fields of RESOURCE_ID, RESOURCE_TYPE and
+        RESOURCE_PROPERTIES
+        :return: dict resource_deploy_info- A dictionary of deployment
+        information of the resource to be stored at deploy time. It includes the resource id of
+        the resource along with the physical id and deploy output of the resource.
+        """
+        try:
+            resource_id = resource[RESOURCE_ID]
+            resource_type = resource[RESOURCE_TYPE]
+            resource_properties = resource[RESOURCE_PROPERTIES]
+        except KeyError as e:
+            raise StackError("%s doesn't exist in resource config" % str(e))
+
+        # Deployment
+        physical_id = self._get_deployed_resource(resource_id, resource_type)
+
+        if resource_type == JOBS_TYPE:
+            physical_id, deploy_output = self.deploy_job(resource_id, resource_properties,
+                                                         physical_id)
+        else:
+            raise StackError("Resource type '%s' not found" % resource_type)
+
+        resource_deploy_info = {RESOURCE_ID: resource_id, RESOURCE_TYPE: resource_type}
+        if six.PY3:
+            resource_deploy_info[RESOURCE_DEPLOY_TIMESTAMP] = datetime.now().timestamp()
+        elif six.PY2:
+            resource_deploy_info[RESOURCE_DEPLOY_TIMESTAMP] = \
+                time.mktime(datetime.now().timetuple())
+        resource_deploy_info[RESOURCE_PHYSICAL_ID] = physical_id
+        resource_deploy_info[RESOURCE_DEPLOY_OUTPUT] = deploy_output
+        return resource_deploy_info
+
+    def deploy(self, filename):  # overwrite to be added
+        """
+        Deploys a stack given stack JSON configuration template at path filename.
+
+        Loads the JSON template as well as status JSON if stack has been deployed before.
+        After going through each of the resources and deploying them, stores status JSON
+        of deployment with deploy status of each resource deployment.
+
+        :param filename: Path to stack JSON configuration template
+        :return: None.
+        """
+        config_filepath = os.path.abspath(filename)
+        config_dir = os.path.dirname(config_filepath)
+        cli_cwd = os.getcwd()
+        os.chdir(config_dir)  # Switch current working directory to where json is stored
+        try:
+            parsed_conf = self._parse_config_file(filename)
+            stack_name = parsed_conf['name'] if 'name' in parsed_conf else None
+
+            self._load_deploy_metadata(config_filepath)
```

Comment:
As I mentioned above, it would be cleaner to generate `status_path` using file path and simply use _parse_config_file to parse the json.